### PR TITLE
Data Explorer: Generate download script

### PR DESCRIPTION
**Pull request by justinegeffen**
Addresses https://github.com/seqeralabs/platform/issues/6644 and https://github.com/seqeralabs/tower-backlog/issues/1076

### CODE REVIEW

File: platform_versioned_docs/version-23.4.0/data/data-explorer.mdx [hunk 1]
```diff
@@ -92,6 +88,10 @@ Data Explorer supports file uploads to your private cloud storage buckets. From
 You must configure cross-origin resource sharing (CORS) for your cloud storage provider to allow file uploads from Seqera. CORS configuration differs for each provider.
 :::
 
+### Download multiple files using a script
+
+If you are a Data Studios private preview customer, you can download multiple files either via the user interface or with a script.
```
Comment (alvaromartmart) [inline platform_versioned_docs/version-23.4.0/data/data-explorer.mdx:93]: this feature is not related to Data Studios
```suggestion
You can download multiple files either via the user interface or with a script.
```

**Review by alvaromartmart — CHANGES_REQUESTED** [review]

File: platform_versioned_docs/version-23.4.0/data/data-explorer.mdx [hunk 2]
```diff
@@ -88,6 +88,18 @@ Data Explorer supports file uploads to your private cloud storage buckets. From
 You must configure cross-origin resource sharing (CORS) for your cloud storage provider to allow file uploads from Seqera. CORS configuration differs for each provider.
 :::
 
+### Download multiple files
+
+You can download multiple files via the user interface or using a script in your cloud provider's CLI. 
+
+Open the cloud bucket that you'd like to download and select the download icon in the **Actions** column. You can choose **Download selected** or **Download selected using code**. The code snippet provided is specific to the cloud provider you've configured. You may be prompted to authenticate during the download process.
+
+Refer to your cloud provider's documentation for troubleshooting credential-related issues:
```
Comment (alvaromartmart) [inline platform_versioned_docs/version-23.4.0/data/data-explorer.mdx:97]:
```suggestion
Open the cloud bucket that you'd like to download and select the download icon in the **Actions** column. You can choose **Download selected** or **Download selected using code**. 

The code snippet provided is specific to the cloud provider you've configured. You may be prompted to authenticate during the download process. Refer to your cloud provider's documentation for troubleshooting credential-related issues:
```
Mainly because it seems likely that we'll need to add a separate paragraph to explain the other option ("Download selected"), to which the provider cli instructions don't apply

**Review by alvaromartmart — CHANGES_REQUESTED** [review]

File: platform_versioned_docs/version-23.4.0/data/data-explorer.mdx (hunk 2 above)
Comment (justinegeffen) [reply platform_versioned_docs/version-23.4.0/data/data-explorer.mdx:97]: Good point. This will be added when I can test out the feature.

**Comment by jason-seqera** [comment]
Refs https://github.com/seqeralabs/tower-backlog/issues/1270

File: platform_versioned_docs/version-23.4.0/data/data-explorer.mdx [hunk 3]
```diff
@@ -88,6 +88,22 @@ Data Explorer supports file uploads to your private cloud storage buckets. From
 You must configure cross-origin resource sharing (CORS) for your cloud storage provider to allow file uploads from Seqera. CORS configuration differs for each provider.
 :::
 
+### Download multiple files
+
+You can download multiple files via the user interface or using a script in your cloud provider's CLI. 
+
+:::note
```
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-explorer.mdx:95]: I don't think this is correct. Downloading multiple folders/files works, but the folder structure is not preserved in non-Chromium based browsers.

File: platform_versioned_docs/version-23.4.0/data/data-explorer.mdx [hunk 4]
```diff
@@ -88,6 +88,22 @@ Data Explorer supports file uploads to your private cloud storage buckets. From
...
+The browser download functionality currently does not work in non-Chromium browsers.
+:::
+
+Open the cloud bucket that you'd like to download and select the download icon in the **Actions** column. You can choose **Download selected** or **Download selected using code**.
```
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-explorer.mdx:99]: See [Slack discussion](https://seqera.slack.com/archives/C06B6RRGVCZ/p1714411320345289) on removing the `Actions` column

Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-explorer.mdx:99]:
```suggestion
Open the cloud bucket that you'd like to download files and folder from. By default, download the contents of the current directory by choosing **Download current directory**. Alternatively, use checkboxes to select specific files and folders, and choose the **Download** button: you can **Download files** via the browser or **Download using code**. 
```

File: platform_versioned_docs/version-23.4.0/data/data-explorer.mdx [hunk 5]
```diff
@@ -88,6 +88,22 @@ Data Explorer supports file uploads to your private cloud storage buckets. From
 You must configure cross-origin resource sharing (CORS) for your cloud storage provider to allow file uploads from Seqera. CORS configuration differs for each provider.
 :::
 
+### Download multiple files
```
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-explorer.mdx:91]:
```suggestion
## Download multiple files
```

File: platform_versioned_docs/version-23.4.0/data/data-explorer.mdx [hunk 6]
```diff
@@ -88,6 +88,22 @@ Data Explorer supports file uploads to your private cloud storage buckets. From
...
+
+Open the cloud bucket that you'd like to download and select the download icon in the **Actions** column. You can choose **Download selected** or **Download selected using code**. 
+
+The code snippet provided is specific to the cloud provider you've configured. You may be prompted to authenticate during the download process. Refer to your cloud provider's documentation for troubleshooting credential-related issues:
+
+- [GCP](https://cloud.google.com/sdk/gcloud/reference/storage)
+- [AWS](https://docs.aws.amazon.com/cli/latest/reference/s3/)
+- [Azure](https://learn.microsoft.com/en-us/cli/azure/storage?view=azure-cli-latest)
+
 ### AWS S3 CORS configuration
```
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-explorer.mdx:107]: There needs to be a CORS policy applied for multiple file downloads too. See the validation [tests](https://github.com/seqeralabs/tower-backlog/issues/1270).

**Comment by jason-seqera** [comment]
@justinegeffen, I rebased this against the indentation fixes for this, which required a force-push, so that unfortunately de-syncs your local branch. I'm going to wrap this PR up, so that's probably okay going forward. But if you need to use the branch locally, you'll need to `branch -m de-file-download de-file-download.old` and then `get fetch origin` and `git checkout de-file-download` to get the re-based branch. (You can always delete the `de-file-download.old` branch as well if it isn't necessary.)

File: platform_versioned_docs/version-23.4.0/data/data-explorer.mdx (hunk 3 above)
Comment (jason-seqera) [reply platform_versioned_docs/version-23.4.0/data/data-explorer.mdx:95]: @robnewman 

> FF/Safari download of folders doesn't preserve the folder-file structure but instead concatenates the folder hierarchy and file name. This is an acceptable work around for now.

What does the concatenated filename look like? Thanks!

File: platform_versioned_docs/version-23.4.0/data/data-explorer.mdx (hunk 6 above)
Comment (jason-seqera) [reply platform_versioned_docs/version-23.4.0/data/data-explorer.mdx:107]: @robnewman, is this a request to incorporate those steps here? Or just to note that a user needs to deal with CORS policies somehow? Thanks!

**Comment by jason-seqera** [comment]
@robnewman, added: https://github.com/seqeralabs/tower-backlog/issues/1076#issuecomment-2070957936

File: platform_versioned_docs/version-23.4.0/data/data-explorer.mdx (hunk 6 above)
Comment (robnewman) [reply platform_versioned_docs/version-23.4.0/data/data-explorer.mdx:107]: It would be helpful to have examples for sure.

**Comment by jason-seqera** [comment]
The CORS content for this is pending on Rob contemplating further what content might make sense to include.

File: platform_versioned_docs/version-23.4.0/data/data-explorer.mdx [hunk 7]
```diff
@@ -88,6 +88,22 @@ Data Explorer supports file uploads to your private cloud storage buckets. From
 You must configure cross-origin resource sharing (CORS) for your cloud storage provider to allow file uploads from Seqera. CORS configuration differs for each provider.
 :::
 
+## Download multiple files
+
+You can download multiple files, up to a maximum of 1,000 files, from the user interface or using a script in your cloud provider's CLI.
```
Comment (alvaromartmart) [inline platform_versioned_docs/version-23.4.0/data/data-explorer.mdx:93]: this is not correct, the 1000 files limit doesn't apply to the "download with script" flow, only to the download through browser / UI flow

**Review by alvaromartmart — CHANGES_REQUESTED** [review]

File: platform_versioned_docs/version-23.4.0/data/data-explorer.mdx (hunk 7 above)
Comment (jason-seqera) [reply platform_versioned_docs/version-23.4.0/data/data-explorer.mdx]: I've updated this, thanks!

**Comment by robnewman** [comment]
@justinegeffen We need to wait until Wednesday

**Comment by jason-seqera** [comment]
@robnewman, I've applied these changes, thanks!

**Review by robnewman — APPROVED** [review]

File: platform_versioned_docs/version-23.4.0/data/data-explorer.mdx [hunk 8]
```diff
@@ -88,9 +88,29 @@ Data Explorer supports file uploads to your private cloud storage buckets. From
 You must configure cross-origin resource sharing (CORS) for your cloud storage provider to allow file uploads from Seqera. CORS configuration differs for each provider.
 :::
 
-### AWS S3 CORS configuration
+## Download multiple files
 
-Apply a [CORS configuration](https://docs.aws.amazon.com/AmazonS3/latest/userguide/ManageCorsUsing.html) to enable file uploads from Seqera to specific S3 buckets. The CORS configuration is a JSON file that defines the origins, headers, and methods allowed for resource sharing requests to a bucket. Follow [these AWS instructions](https://docs.aws.amazon.com/AmazonS3/latest/userguide/enabling-cors-examples.html) to apply the CORS configuration below to each bucket you wish to enable file uploads for:
+You can download up to a maximum of 1,000 files using the browser interface, or an unlimited number of files with the auto-generated download script that uses your cloud provider's CLI and credentials.
```
Comment (alvaromartmart) [inline platform_versioned_docs/version-23.4.0/data/data-explorer.mdx:93]: small note: this `1000` limit is configurable. Its currently set to that value in theory it would be possible to change it. Not sure if these kind of details are usually included in the docs

**Review by alvaromartmart — APPROVED** [review]

File: platform_versioned_docs/version-23.4.0/data/data-explorer.mdx (hunk 8 above)
Comment (justinegeffen) [reply platform_versioned_docs/version-23.4.0/data/data-explorer.mdx:93]: We can certainly add a note indicating that this is a configurable limit but then we'd need to have some sort of maximum value. It's not ideal to say it's adjustable without knowing what the upper limit is, imo.

I propose we let support know about this in case customers want to change the value. @alvaromartmart, where is this value found so that it can be changed?

Comment (justinegeffen) [reply platform_versioned_docs/version-23.4.0/data/data-explorer.mdx:93]: This can be a separate PR/discussion btw. 

Tagging @starnsd for visibility into this thread.

Comment (justinegeffen) [reply platform_versioned_docs/version-23.4.0/data/data-explorer.mdx:93]: Created https://seqera.atlassian.net/browse/EDU-161 to track this and resolving this conversation so we can merge.
